### ISSUE TITLE
Compiling the schema does not preserve the base URI

### ISSUE THREAD

**Issue by ndw** [schxslt/schxslt#217]
I have some code that compiles the schema and then runs a transformation with the compiled schema. That works fine in 1.6.2. Here's the schema in question, though I don't think the exact schema is relevant.

```
      <s:schema queryBinding="xslt2"
                xmlns:s="http://purl.oclc.org/dsdl/schematron"
                xmlns="http://www.w3.org/1999/xhtml">
         <s:pattern>
            <s:rule context="/">
               <s:assert test="deep-equal(., doc('../documents/docbook-valid.xml'))">The document is not equal to the source document.</s:assert>
            </s:rule>
         </s:pattern>
      </s:schema>
```

In 1.8.4, the `doc` function fails to find the document because the base URI of the `s:assert` element has become the static base URI of the compiler stylesheet (...`/2.0/compile/compile-2.0.xsl`)

There are some suspicious looking changes around lines 297 and 315 (in 1.8.4) where previous references to a `$baseUri` variable were removed. I tried putting them back, but didn't have immediate success so I could be completely mistaken.

**Comment by dmj** [comment]
To clarify: There are no @xml:base attributes in the validation stylesheet, are there?

I summed up my findings about relative URIs and base URI in #179 -- The relative URI reference in your case is resolved to whatever the base URI of the validation stylesheet is.

**Comment by ndw** [comment]
Ok, so it's a feature, not a bug. I find it wholly unsatisfying, however. The author of the schema has no possible way of knowing what the location of the compiling stylesheet will be. If I have a project that includes some Schematron schemas and I want to refer to documents in the project, what are my options?
1. Make all the URI references absolute. That's unsatisfying because it means you can't use my project unless you store it in `/home/norm/Projects/myproject`.
2. Play funny games with a URI resolver. That's unsatisfying because it requires setting that up.
3. Don't use URIs. That's just unsatisfying.

In the short term, my hack is to rewrite the compiled schema changing its base URI back to the base URI of the original schema.

**Comment by dmj** [comment]
> The author of the schema has no possible way of knowing what the location of the compiling stylesheet will be.

True. And the compiler has no possible way of knowing where the validation stylesheet will be used. v1.6.2 assumed that the validation stylesheet will be used in the same location like the schema, but this is not always the case.

I consider adding a compiler option but I haven't tought it through wrt the base URI of included items.

**Comment by dmj** [comment]
I changed the title to 'Compiling the schema does not preserve the base URI' bc I find it less misleading. SchXslt does not change the base URI. It doesn't preserve it -- I think handling the base URI of the compiled stylesheet is out of scope of the SchXslt compiler, the problem should be adressed by the program that executes the transformations.

In SchXslt Java I set the document URI of the compiled stylesheet to the URI of the schema:

https://github.com/schxslt/schxslt-java/blob/bf2a685a178e8a2998bb1b24c57d26858cfec225/src/main/java/name/dmaus/schxslt/Compiler.java#L93
